src/js/modules/articles/components/ArticleBody.js

```diff
@@ -93,42 +108,29 @@ const styles = {
   },
 };
 
-const ArticleBody = ({
-  classes,
-  article: { content, title },
-  articles,
-  media,
-}) => {
+const ArticleBody = ({ classes, data, article }) => {
   const isCarouselButtonVisible =
-    SPEC_IMG_CAROUSEL_PATTERN.test(content) && Object.values(media).length > 0;
-  const referencedArticleId = SPEC_REFERENCE_PATTERN.test(content)
-    ? parseInt(SPEC_REFERENCE_PATTERN.exec(content)[1])
-    : null;
-  let referencedArticle = null;
-  if (referencedArticleId) {
-    referencedArticle = articles.find(
-      article => article.id === referencedArticleId,
-    );
-  }
+    SPEC_IMG_CAROUSEL_PATTERN.test(article.content) && Object.values(media).length > 0;
+
   return (
     <Row>
       <Col xs={12} sm={12} md={8} lg={8} className={classes.ArticleBody}>
-        {SPEC_IMG_CAROUSEL_PATTERN.test(content) && (
-          <Lightbox title={title}>
-            <Gallery media={Object.values(media)} />
+        {SPEC_IMG_CAROUSEL_PATTERN.test(article.content) && (
+          <Lightbox title={article.title}>
+            <Gallery media={article.media} />
           </Lightbox>
         )}
-        {Object.values(media).length > 0 && (
+        {article.media.length > 0 && (
```

Comment:
Ugh this whole bunch of code is just awkward. You have like 2 and a half situations here. Media is greater than zero but no spec carousel pattern, spec carousel pattern but no media, spec carousel pattern and exactly one media, etc. It's just...hard to reason about